File: PocketBeagle-2/seven_segment/rust/src/main.rs
use std::io::Write;
use std::thread::sleep;
use std::time::Duration;

use beagle_helper::sysfs::Device;

pub const SEVEN_SEGMENT_LEFT: &str = "/sys/devices/platform/seven-segments-left/linedisp.1/";
pub const SEVEN_SEGMENT_RIGHT: &str = "/sys/devices/platform/seven-segments-right/linedisp.0/";

fn main() {
    let segment_left = Device::with_path(SEVEN_SEGMENT_LEFT).unwrap();
    let segment_right = Device::with_path(SEVEN_SEGMENT_RIGHT).unwrap();

    let mut left_msg = segment_left.sysfs_w("message").unwrap();
    let mut right_msg = segment_right.sysfs_w("message").unwrap();

    segment_left
        .sysfs_w("scroll_step_ms")
        .unwrap()
        .write_string(1000)
        .unwrap();
    segment_right
        .sysfs_w("scroll_step_ms")
        .unwrap()
        .write_string(1000)
        .unwrap();

    println!("Countdown");
    left_msg.write_all(b"10000000000").unwrap();
    right_msg.write_all(b"09876543210").unwrap();

    sleep(Duration::from_secs(11));

    left_msg.write_all(b" ").unwrap();
    right_msg.write_all(b" ").unwrap();
}

File: PocketBeagle-2/seven_segment/python/main.py
from time import sleep
from sysfs import Device
from pathlib import Path

TECHLAB_SEVEN_SEGMENT_LEFT = Path("/sys/devices/platform/seven-segments-left/linedisp.1/")
TECHLAB_SEVEN_SEGMENT_RIGHT = Path("/sys/devices/platform/seven-segments-right/linedisp.0/")

segment_left = Device(path=TECHLAB_SEVEN_SEGMENT_LEFT)
segment_right = Device(path=TECHLAB_SEVEN_SEGMENT_RIGHT)

left_msg = segment_left.sysfs('message')
right_msg = segment_right.sysfs('message')

segment_left.sysfs('scroll_step_ms').write(1000)
segment_right.sysfs('scroll_step_ms').write(1000)

print("Countdown")
left_msg.write('10000000000')
right_msg.write('09876543210')

sleep(11)

left_msg.write(' ')
right_msg.write(' ')
